### PR TITLE
Block lists

### PR DESCRIPTION
**Pull request by devinivy**
Spruced-up #967 as an implementation of blocklists.

### CODE REVIEW

File: packages/pds/src/db/migrations/20230524T211916770Z-block-lists.ts
Comment (dholms) [inline packages/pds/src/db/migrations/20230524T211916770Z-block-lists.ts:23]: ah we realized we didn't need this cause of the unique constraint

**Review by dholms — APPROVED** [review]
sweet yup this LGTM 👍

**Comment by dholms** [comment]
before merge, can we get https://github.com/bluesky-social/atproto/pull/980 in & make sure this PR is up to speed with the tightening?

**Comment by temujin9** [comment]
Hey guys, this has been sitting at "LGTM but needs rebase" for a while now. Any chance I could encourage you to rebase and merge soon?

**Comment by dholms** [comment]
Hah it has been awhile hasn't it 😅 

We put this together somewhat naively & are going to hold off on releasing it for now for two main reasons:
- there's some concern about the indexing impact of a large fanout from blocklists. it could add quite a bit of overhead on either the read or the write side
- there's also concern about the social impact of blocklists. research is fairly mixed on if their impact is positive or negative. I still think they're an important primitive, but we want to put some work into other moderation primitives (composable labeling) before releasing blocklists

I'm actually going to close this for now as it's misleading in it's current state. I won't delete the branch tho since I'm hoping to use this code soonish

**Comment by temujin9** [comment]
As predicted, https://bskyblocklist.com/ is now a thing. I would really prefer a native tool. Can discussion on this be resurrected?

**Comment by dholms** [comment]
Yup the idea hasn't been abandoned - we actually discussed blocklists today & we're planning on fleshing out the use cases of lists more broadly here soon. Can't promise a timeline, but hopefully coming back to this somewhat soon

**Pull request closed without merging**
